### ISSUE TITLE
KGQAn Server startup issue

### ISSUE THREAD

**Issue by DamianMYM** [CoDS-GCS/KGQAn#6]
Hello Reham,

I still can not initiate the KGQAn server I really appreciate your work and I hope I can run it successfully. (I started a new one due to a long time waiting for your response lol).

This time I set up the correct path and started kgqan server again, it could load the model properly.
**BUT,** I had been waiting for like 1h and nothing shown on the terminal after automately downloading everything, which I should have expected this message: Server started http://0.0.0.0:8899
<img width="879" alt="image" src="https://github.com/CoDS-GCS/KGQAn/assets/113518689/f7f7ebf0-aa25-4638-b227-6cd9a6ddfc41">

<img width="884" alt="image" src="https://github.com/CoDS-GCS/KGQAn/assets/113518689/24ff2fbf-ba01-46e7-924f-c09e26c582e3">
I do have another terminal operating the word-embedding server simultaneously like this.
<img width="885" alt="image" src="https://github.com/CoDS-GCS/KGQAn/assets/113518689/59c5517f-e7c7-48e5-b2df-e44204b8fc69">

This is really strange and I have no idea how to solve it.

Looking forward to your reply.

Many thanks,
Damian

**Comment by Reham-Osama** [comment]
Hi Damian,

Sorry for the late reply, Can you please pull the latest changes, It was a logging error that should be fixed now

Please let me know if you are facing any other errors.

**Comment by DamianMYM** [comment]
Hello Reham,

Thanks for your patient reply!

One last question for server mode, I successfully launched the server, and sent the request both by CURL and POSTMAN followed by the instructions from your home page.
Unfortunately, there is no response at all. 

Due to other works, I tried this today and found the issue. I am really into your work and sorry for so many questions.

Many thanks,
Damian

**Comment by Reham-Osama** [comment]
Hello Damian,

Can you please send me the curl request you are sending, and the server log corresponding to this request?

**Comment by DamianMYM** [comment]
`curl -X POST -H "Content-Type: application/json" \
     -d '{"question": "Who founded Intel?", "knowledge_graph": "dbpedia", "max_answers": 3}' \
     http://0.0.0.0:8899`
     Simply followed your example. And unfortunately, there is nothing about this request.

**Comment by Reham-Osama** [comment]
When you check the server logs themselves, i.e. the window where it prints `Server started http://0.0.0.0:8899`
Are there any lines that follow this line?

**Comment by DamianMYM** [comment]
No, nothing, only printed this line, and then nothing returned.
`Server started http://0.0.0.0:8899`

**Comment by Reham-Osama** [comment]
Hi Damian,
I tested a new installation of the system, and followed the steps and it worked for me returning this output
`[{"question": "Who founded Intel?", "sparql": " SELECT ?var1 ?type WHERE {    <http://dbpedia.org/resource/Intel> <http://dbpedia.org/property/founders> ?var1 .     OPTIONAL {       ?var1 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> ?type .     } }", "values": ["http://dbpedia.org/resource/Robert_Noyce", "http://dbpedia.org/resource/Gordon_Moore"], "named_entites": ["Intel", "var1"], "extracted_relation": [["Intel", "var1", "founded"]], "score": 0.771619810277574, "nodes": ["http://dbpedia.org/resource/Intel"], "edges": ["http://dbpedia.org/property/founders"]}, {"question": "Who founded Intel?", "sparql": " SELECT ?var1 ?type WHERE {    ?var1 <http://dbpedia.org/ontology/knownFor> <http://dbpedia.org/resource/Intel> .     OPTIONAL {       ?var1 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> ?type .     } }", "values": ["http://dbpedia.org/resource/Andrew_Grove", "http://dbpedia.org/resource/Gordon_Moore", "http://dbpedia.org/resource/Arthur_Rock"], "named_entites": ["Intel", "var1"], "extracted_relation": [["Intel", "var1", "founded"]], "score": 0.6947743530520285, "nodes": ["http://dbpedia.org/resource/Intel"], "edges": ["http://dbpedia.org/ontology/knownFor"]}]`

For your case, it seems that the request does not reach the server, which maybe is because of a network error, Can you make sure that the machine you are using has port 8899 open? If not, you can change the port number in `src.kgqan.server.py` Line 12 for another port that will work with your machine

**Comment by DamianMYM** [comment]
![image](https://github.com/CoDS-GCS/KGQAn/assets/113518689/7296b786-8fdc-4e99-acb9-e87304f2b3c6)

Hi Reham,

Thanks for your reply, though as the image shows my instance is still not working.

Since received your message, here are the things I tried.
1. I checked the firewall, which is closed definitely. Also, the network works fine.
2. And then I tried another port, which can work properly while using the ssh command to communicate with my machine. BUT still.
3. I checked the log, and there is still nothing in there.

May I ask if it could be working by batching inputs perhaps? As I haven't tried the other way. And I feel so sorry for taking the time to answer my questions. 

Many thanks,
Damian

**Comment by Reham-Osama** [comment]
Hi Damian,

Can you try this command and let me know the result `telnet 0.0.0.0 8899`

**Comment by DamianMYM** [comment]
![image](https://github.com/CoDS-GCS/KGQAn/assets/113518689/bfd31115-bb73-43eb-8453-0b4b4fcec095)
 
I can see that  It has connected.

**Comment by Reham-Osama** [comment]
Hi Damian, 
This is so strange, Can you try restarting the server and see if the error persists ?
